### PR TITLE
[wasm] windows build improvements

### PR DESCRIPTION
**Pull request by radekdoulik**
Fix build on windows in folders containing space(s) in the name and also
in case where `EMSDK_PATH` contains space(s).

Use immediate set for `EMSCRIPTEN_VERSION` to read `emscripten-version.txt`
file only once.

### CODE REVIEW

File: src/mono/wasm/Makefile [hunk 1]
```diff
@@ -33,7 +33,7 @@ all: build-native icu-files source-files header-files
 #  If EMSDK_PATH is not set by the caller, download and setup a local emsdk install.
 #
 
-EMSCRIPTEN_VERSION=`cat emscripten-version.txt`
+EMSCRIPTEN_VERSION:=`cat emscripten-version.txt`
```
Comment (kjpou1) [inline src/mono/wasm/Makefile:36]: I ran into another problem with the file read as outlined in the issue here: https://github.com/dotnet/runtime/issues/49862

Sent a PR: https://github.com/dotnet/runtime/pull/49863

**Comment by radical** [comment]
I had to quote paths in a few more places:

```diff
diff --git a/src/libraries/Common/src/System/Security/Cryptography/Asn1/AsnXml.targets b/src/libraries/Common/src/System/Security/Cryptography/Asn1/AsnXml.targets
index 751187ddfe6..a923cdca87c 100644
--- a/src/libraries/Common/src/System/Security/Cryptography/Asn1/AsnXml.targets
+++ b/src/libraries/Common/src/System/Security/Cryptography/Asn1/AsnXml.targets
@@ -29,13 +29,13 @@
     <Exec
       IgnoreExitCode="true"
       StandardOutputImportance="Low"
-      Command="$(_AsnXmlDiffCmd) @(AsnXml -> '$(_AsnIntermediatePath)%(filename).cs') @(AsnXml -> '%(Identity).cs')">
+      Command="$(_AsnXmlDiffCmd) @(AsnXml -> '&quot;$(_AsnIntermediatePath)%(filename).cs&quot;') @(AsnXml -> '&quot;%(Identity).cs&quot;')">
       <Output TaskParameter="ExitCode" ItemName="_AsnXmlDiffCode" />
     </Exec>
 
     <Copy
       Condition="'@(_AsnXmlDiffCode)' != '0'"
-      SourceFiles="@(AsnXml -> '$(_AsnIntermediatePath)%(filename).cs')"
+      SourceFiles="@(AsnXml -> $(_AsnIntermediatePath)%(filename).cs')"
       DestinationFiles="@(AsnXml -> '%(Identity).cs')" />
 
     <Warning Condition="'@(_AsnXmlDiffCode)' != '0'" Text="AsnXml regenerated files, be sure to check them in: @(AsnXml -> '%(Identity).cs')" />
diff --git a/src/mono/mono.proj b/src/mono/mono.proj
index 6194144eaea..fbd04bb7400 100644
--- a/src/mono/mono.proj
+++ b/src/mono/mono.proj
@@ -270,7 +270,7 @@
       <_MonoCMakeArgs Include="-DENABLE_LLVM_RUNTIME=1"/>
       <_MonoCFLAGS Include="-fexceptions"/>
       <_MonoCXXFLAGS Include="-fexceptions"/>
-      <_MonoCFLAGS Include="-I$([MSBuild]::NormalizePath('$(PkgMicrosoft_NETCore_Runtime_ICU_Transport)', 'runtimes', 'browser-wasm', 'native', 'include'))"/>
+      <_MonoCFLAGS Include="\&quot;-I$([MSBuild]::NormalizePath('$(PkgMicrosoft_NETCore_Runtime_ICU_Transport)', 'runtimes', 'browser-wasm', 'native', 'include'))\&quot;"/>
     </ItemGroup>
     <!-- iOS/tvOS specific options -->
     <PropertyGroup Condition="'$(TargetsiOS)' == 'true' or '$(TargetstvOS)' == 'true'">
```

And with that added on this PR, I can successfully build on windows!!

**Review by AntonLapounov — APPROVED** [review]
Looks good to me. Thank you for fixing this.

File: eng/native/init-vs-env.cmd [hunk 2]
```diff
@@ -68,4 +68,4 @@ set "__VCBuildArch="
 
 :: Set CMakePath by evaluating the output from set-cmake-path.ps1.
 :: In case of a failure the output is "exit /b 1".
-for /f "delims=" %%a in ('powershell -NoProfile -ExecutionPolicy ByPass "%~dp0set-cmake-path.ps1"') do %%a
+for /f "delims=" %%a in ('powershell -NoProfile -ExecutionPolicy ByPass "& ""%~dp0set-cmake-path.ps1"""') do %%a
```
Comment (AntonLapounov) [inline eng/native/init-vs-env.cmd:71]: Could we use `-File "%~dp0set-cmake-path.ps1"` instead?

Comment (radekdoulik) [reply eng/native/init-vs-env.cmd]: That works too and looks better.

**Comment by radekdoulik** [comment]
> 
> I had to quote paths in a few more places:
> 
> ...
> 
> And with that added on this PR, I can successfully build on windows!!

Thanks! I added commit with your changes. I just left out this one, which looked weird and hopefully not needed?

    -      SourceFiles="@(AsnXml -> '$(_AsnIntermediatePath)%(filename).cs')"
    +      SourceFiles="@(AsnXml -> $(_AsnIntermediatePath)%(filename).cs')"

File: src/libraries/Common/src/System/Security/Cryptography/Asn1/AsnXml.targets [hunk 3]
```diff
@@ -29,13 +29,13 @@
...
       StandardOutputImportance="Low"
-      Command="$(_AsnXmlDiffCmd) @(AsnXml -> '$(_AsnIntermediatePath)%(filename).cs') @(AsnXml -> '%(Identity).cs')">
+      Command="$(_AsnXmlDiffCmd) @(AsnXml -> '&quot;$(_AsnIntermediatePath)%(filename).cs&quot;') @(AsnXml -> '&quot;%(Identity).cs&quot;')">
       <Output TaskParameter="ExitCode" ItemName="_AsnXmlDiffCode" />
     </Exec>
 
     <Copy
       Condition="'@(_AsnXmlDiffCode)' != '0'"
-      SourceFiles="@(AsnXml -> '$(_AsnIntermediatePath)%(filename).cs')"
+      SourceFiles="@(AsnXml -> $(_AsnIntermediatePath)%(filename).cs')"
```
Comment (AntonLapounov) [inline src/libraries/Common/src/System/Security/Cryptography/Asn1/AsnXml.targets:38]: Is this correct?

Comment (kjpou1) [reply src/libraries/Common/src/System/Security/Cryptography/Asn1/AsnXml.targets:38]: looks like a ' was removed by accident?

Comment (radekdoulik) [reply src/libraries/Common/src/System/Security/Cryptography/Asn1/AsnXml.targets]: Doh, I left it out first, but then overlooked it when adding other changes.

File: src/mono/mono.proj [hunk 4]
```diff
@@ -261,7 +271,7 @@
       <_MonoCMakeArgs Include="-DENABLE_LLVM_RUNTIME=1"/>
       <_MonoCFLAGS Include="-fexceptions"/>
       <_MonoCXXFLAGS Include="-fexceptions"/>
-      <_MonoCFLAGS Include="-I$([MSBuild]::NormalizePath('$(PkgMicrosoft_NETCore_Runtime_ICU_Transport)', 'runtimes', 'browser-wasm', 'native', 'include'))"/>
+      <_MonoCFLAGS Include="$(EscapedQuoteW)-I$([MSBuild]::NormalizePath('$(PkgMicrosoft_NETCore_Runtime_ICU_Transport)', 'runtimes', 'browser-wasm', 'native', 'include'))$(EscapedQuoteW)"/>
```
Comment (AntonLapounov) [inline src/mono/mono.proj:274]: What happens on Linux if the path contains spaces?

**Pull request merged**